src/core/main.go

```diff
@@ -74,6 +79,32 @@ func updateInitPassword(userID int, password string) error {
 	return nil
 }
 
+// Quota migration
+func quotaSync() error {
+	usages, err := dao.ListQuotaUsages()
+	if err != nil {
+		log.Errorf("list quota usage error, %v", err)
+		return err
+	}
+	projects, err := dao.GetProjects(nil)
+	if err != nil {
+		log.Errorf("list project error, %v", err)
+		return err
+	}
+
+	// upgrade from old version
+	if len(projects) > 1 && len(usages) == 1 {
```

Comment:
How can we determine it's a upgrade version or fresh installation?

Reply:
Per our discussion offline, update the migration sql and insert quota record for `library` only

Reply:
will enhance the condition per discussion.